### PR TITLE
Feat(web-react): Introduce FieldGroup component #DS-856

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

Introduce `FieldGroup` component in react.

### Additional context

- [x] Added react demo
- [x] Added story to storybook 
- [x] Tests

### Issue reference
[DS-856 web-react: FieldGroup komponenta](https://jira.lmc.cz/browse/DS-856)

### CODE REVIEW

File: packages/web-react/src/components/FieldGroup/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# FieldGroup
+
+FieldGroup is a component that groups form fields together. Additionally, it provides a common label, helper text, and
```
Comment (literat) [inline packages/web-react/src/components/FieldGroup/README.md:3]: Can we have every sentence on a new line, please? For better future diffs ;-)

File: packages/web-react/src/components/FieldGroup/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+| `form`                  | `string`                                       | `null`  | no       | Parent form ID                             |
+| `id`                    | `string`                                       | —       | yes      | Group and label identification             |
+| `isDisabled`            | `bool`                                         | `false` | no       | If true, the group is disabled             |
+| `isLabelHidden`         | `bool`                                         | `false` | no       | If true, label is hidden                   |
+| `isRequired`            | `bool`                                         | `false` | no       | If true, the group is marked as required   |
+| `label`                 | `string`                                       | `null`  | no\*     | Label text                                 |
+| `name`                  | `string`                                       | `null`  | no       | Group name                                 |
+| `UNSAFE_helperText`     | `string`                                       | `null`  | no\*\*   | Unescaped custom helper text (allows HTML) |
```
Comment (literat) [inline packages/web-react/src/components/FieldGroup/README.md:107]: We do not have `UNSAFE_` props here. I guess this is a copy paste from **web-twig** ;-)

File: packages/web-react/src/components/FieldGroup/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+| `UNSAFE_label`          | `string`                                       | `null`  | no\*     | Unescaped label text (allows HTML)         |
+| `UNSAFE_validationText` | `string`, `string[]`                           | `null`  | no\*\*   | Unescaped validation text (allows HTML)    |
+| `validationState`       | [Validation dictionary][dictionary-validation] | `null`  | no       | Type of validation state                   |
+| `validationText`        | `string`, `string[]`                           | `null`  | no\*\*   | Validation text                            |
+
+\*: To keep the component accessible, a label is always required. You can use the `label` for simple text or `UNSAFE_label` for HTML content.
+\*\*: Props with and without `UNSAFE_` prefix are mutually exclusive.
+
+On top of the API options, you can add `data-*` or `aria-*` (except `aria-describedby` which is reserved for linking
```
Comment (literat) [inline packages/web-react/src/components/FieldGroup/README.md:116]: This is the same as above.

File: packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    'test-FieldGroup',
+  );
+
+  restPropsTest((props) => <FieldGroup {...props} label="Label" />, 'fieldset');
+
+  it('should render items as children', () => {
+    const dom = render(<FieldGroup label="Label">{itemList}</FieldGroup>);
+
+    const fieldGroup = dom.container.querySelector('fieldset') as HTMLElement;
+    expect(fieldGroup).toHaveClass('FieldGroup');
```
Comment (literat) [inline packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx:37]: I would prefer writing an empty line before `expect` as we do it with the `return` statement. I am suggesting this for consistency with other tests and for better readability.

File: packages/web-react/src/components/FieldGroup/stories/FieldGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// Because there is no `dist` directory during the CI run
```
Comment (literat) [inline packages/web-react/src/components/FieldGroup/stories/FieldGroup.tsx:1]: I think that those comments are useless in this story.

File: packages/web-react/src/types/fieldGroup.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface FieldGroupProps extends FieldGroupElementBaseProps, LabelProps, HelperTextProps, Validation {
+  label: string;
+  isRequired?: boolean;
+  isDisabled?: boolean;
+  isFluid?: boolean;
+  isLabelHidden?: boolean;
+  validationText?: ValidationTextType;
+  validationState?: ValidationState;
+  UNSAFE_label?: string;
```
Comment (literat) [inline packages/web-react/src/types/fieldGroup.ts:20]: Why?

File: packages/web-react/src/types/fieldGroup.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+export type FieldGroupElementBaseProps = SpiritFieldGroupElementPropsWithRef;
+
+export interface FieldGroupProps extends FieldGroupElementBaseProps, LabelProps, HelperTextProps, Validation {
+  label: string;
+  isRequired?: boolean;
+  isDisabled?: boolean;
+  isFluid?: boolean;
+  isLabelHidden?: boolean;
+  validationText?: ValidationTextType;
+  validationState?: ValidationState;
```
Comment (literat) [inline packages/web-react/src/types/fieldGroup.ts:19]: Please, revisit the types. `validationState` is a part of the `Validation` interface.

File: packages/web-react/src/components/FieldGroup/README.md (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/README.md]: done

File: packages/web-react/src/components/FieldGroup/README.md (hunk 2 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/README.md]: removed `UNSAFE_helperText`, `UNSAFE_label` and `UNSAFE_validationText`

File: packages/web-react/src/components/FieldGroup/README.md (hunk 3 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/README.md]: removed

File: packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx (hunk 4 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx:38]: added empty line before, but we still have inconsistency in other tests, somewhere are empty lines, somewhere not.

File: packages/web-react/src/components/FieldGroup/stories/FieldGroup.tsx (hunk 5 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/stories/FieldGroup.tsx]: comments removed

File: packages/web-react/src/types/fieldGroup.ts (hunk 7 above)
Comment (pavelklibani) [reply packages/web-react/src/types/fieldGroup.ts]: Removed `isRequired`, `validationState` and `UNSAFE_label`.
Added `JSX.Element` to be able add HTML tags

File: packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx (hunk 4 above)
Comment (literat) [reply packages/web-react/src/components/FieldGroup/__tests__/FieldGroup.test.tsx:38]: I am ware of that and trying to refactor it.

File: packages/web-react/src/components/FieldGroup/README.md (hunk 1 above)
Comment (literat) [reply packages/web-react/src/components/FieldGroup/README.md]: This was meant for the entire document. But there is a topic for this on our thursday meeting 🥌

**Review by literat — APPROVED** [review]

File: packages/web-react/src/components/FieldGroup/demo/index.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    <DocsSection title="Grouped Checkboxes">
+      <FieldGroup label="Label">
+        <Checkbox id="checkbox0" label="Checkbox Label" name="radioDefault" isChecked />
+        <Checkbox id="checkbox1" label="Checkbox Label" name="radioDefault" />
+      </FieldGroup>
+    </DocsSection>
+    <DocsSection title="Grouped Radio Fields">
+      <FieldGroup label="Label">
+        <Radio id="radio1" label="Radio Label" name="checkboxDefault" isChecked />
+        <Radio id="radio2" label="Radio Label" name="checkboxDefault" />
```
Comment (crishpeen) [inline packages/web-react/src/components/FieldGroup/demo/index.tsx:84]:
```suggestion
        <Radio id="radio2" label="Radio Label" name="radioDefault" />
```

File: packages/web-react/src/types/fieldGroup.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { HelperTextProps, SpiritFieldGroupElementPropsWithRef, Validation, ValidationTextType } from './shared';
+import { LabelProps } from './label';
```
Comment (crishpeen) [inline packages/web-react/src/types/fieldGroup.ts:2]: this also sort by abc pls, label import first and then shared.

File: packages/web-react/src/components/FieldGroup/FieldGroup.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      {...rest}
+    >
+      <legend className="accessibility-hidden">{label}</legend>
+      {!isLabelHidden && (
+        <div className={classProps.label} aria-hidden="true">
+          {label}
+        </div>
+      )}
+      <div className={classProps.fields}>{children}</div>
+      {helperText && <div className={classProps.helperText}>{helperText}</div>}
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/FieldGroup.tsx:46]: As we agreed in the office, there is another issue in the backlog for this: [DS-886](https://jira.lmc.cz/browse/DS-886)

File: packages/web-react/src/components/FieldGroup/README.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
+# FieldGroup
+
+FieldGroup is a component that groups form fields together.
+Additionally, it provides a common label, helper text, and validation messages for all fields in the group.
+
+## Basic Usage
+
+```jsx
+<FieldGroup id="example-field-group" label="Label">
+    <!-- Form fields… -->
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/README.md:10]: changed to jsx/tsx comments 👍

File: packages/web-react/src/components/FieldGroup/demo/index.tsx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    <DocsSection title="Fluid">
+      <FieldGroup label="Label" isFluid>
+        <div className="docs-Box">Item</div>
+        <div className="docs-Box">Item</div>
+        <div className="docs-Box">Item</div>
+      </FieldGroup>
+    </DocsSection>
+    <DocsSection title="Grouped Checkboxes">
+      <FieldGroup label="Label">
+        <Checkbox id="checkbox0" label="Checkbox Label" name="radioDefault" isChecked />
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/demo/index.tsx:77]: fixed

File: packages/web-react/src/components/FieldGroup/demo/index.tsx [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+        <Checkbox id="checkbox1" label="Checkbox Label" name="radioDefault" />
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/demo/index.tsx:78]: fixed

File: packages/web-react/src/components/FieldGroup/demo/index.tsx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+      </FieldGroup>
+    </DocsSection>
+    <DocsSection title="Grouped Radio Fields">
+      <FieldGroup label="Label">
+        <Radio id="radio1" label="Radio Label" name="checkboxDefault" isChecked />
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/demo/index.tsx:83]: fixed

File: packages/web-react/src/components/FieldGroup/demo/index.tsx (hunk 8 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/demo/index.tsx:84]: fixed

File: packages/web-react/src/components/FieldGroup/stories/argTypes.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+export default {
+  validationState: {
+    control: {
+      type: 'select',
+      options: [...Object.values(ValidationStates), undefined],
+    },
+    defaultValue: undefined,
+  },
+  isDisabled: {
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/FieldGroup/stories/argTypes.ts:11]: Added missing props, thanks

File: packages/web-react/src/types/index.ts [hunk 16]
```diff
@@ -28,3 +28,4 @@ export * from './textArea';
 export * from './textField';
 export * from './textFieldBase';
 export * from './tooltip';
+export * from './fieldGroup';
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/types/index.ts:31]: sorted 🔤

File: packages/web-react/src/types/fieldGroup.ts (hunk 9 above)
Comment (pavelklibani) [reply packages/web-react/src/types/fieldGroup.ts:2]: done

**Review by crishpeen — APPROVED** [review]
LGTM

**Pull request merged**